### ISSUE TITLE
Make sure $notoptions is an array

### ISSUE THREAD

**Issue by spacedmonkey** [humanmade/memcache-object-cache#8]
Important before reading the rest of this issue is this - https://core.trac.wordpress.org/ticket/31147

Basically, in the when calling `get_site_option`, it tries to get it out of cache. For some reason it, is returning from cache as something other than a array, which it should be. As the core doesn't do any type checking [here](https://github.com/WordPress/WordPress/blob/master/wp-includes/option.php#L1027), it tries to push something into an variable that isn't an array, causing PHP type errors all over the place. 

I know this issue has nothing to do with this dropin, it is an issue with the core. I was wondering if we could fix this in this dropin, as a work around until this bug is patched and for older sites.

Just an idea but if we did something like this in the get function

``` php
if ( preg_match("\d+:notoptions",$id) && $group === 'site-options' && ! is_array($value) ) { 
   return array();
}
```

It is a little hacky, but might work. Possibly related is [this](https://github.com/humanmade/memcache-object-cache/commit/20f2ae30980a00b0e6e0c3da8d243756664cd950).

Would love the thoughts of @rmccue @joehoyle on this one :smile:

**Comment by rmccue** [comment]
I'm OK with fixing it, _however_ we need to understand why it's returning something other than an array and do full root-cause analysis, rather than "for some reason". If we can confirm what the issue is caused by, let's fix it; if the fix is a workaround in this plugin, happy to workaround while it's patched. :)

**Comment by joehoyle** [comment]
I think core just pushed a fix for this, could be wrong - saw something, somewhere.

**Comment by rmccue** [comment]
Ah, you're right, https://core.trac.wordpress.org/changeset/32943

I'd still like a root cause analysis :)

**Comment by joehoyle** [comment]
Yeah me too, I thought it odd that it was committed without having reproducible steps.

**Comment by felixarntz** [comment]
I found this issue here, related to a core ticket that I had just opened: https://core.trac.wordpress.org/ticket/43506

Essentially, `get_network_option()` should ensure that that cache key is set, otherwise it'll cause countless cache misses and thus will always try to call the external object cache if used. Only a minor optimization, but worth it. Just thought I should post this here, since it is related, because the reason that this value may not be an array is that it simply isn't set everytime it has not been found, which is a bad practice.
